Sources/PromotedCore/Xray.swift

```diff
@@ -208,19 +242,29 @@ fileprivate extension Xray {
   }
   
   private func callDidLog(message: Message) {
+    guard xrayLevel >= .callDetails else { return }
     osLog?.signpostEvent(name: "call", format: "log")
     guard let lastCall = pendingCalls.last else { return }
     lastCall.messages.append(message)
   }
 
   private func callDidError(_ error: Error) {
+    totalErrors += 1
+    if xrayLevel == .batchSummaries {
+      // Create a call and record only the error.
+      let call = Call()
+      call.error = error
+      pendingCalls.append(call)
+      return
+    }
     osLog?.signpostEvent(name: "call", format: "error: %{public}@",
                          error.localizedDescription)
     guard let lastCall = pendingCalls.last else { return }
     lastCall.error = error
   }
 
   private func callDidComplete() {
+    guard xrayLevel >= .callDetails else { return }
```

Comment:
side-comment - Interesting.  I figured the state would still be changed.  I'm happy with this for latency.  Easy to change later if the state is useful.   It'll probably reduce the chance of bugs.